contracts/fungible-tokens/andromeda-merkle-airdrop/src/contract.rs

```diff
@@ -61,6 +55,32 @@ pub fn instantiate(
         },
     )?;
 
+    // IMPORTANT
+    // Permission must be set for Cw20 token
+    // Unless Cw20 is not identified as verified asset
+    if let Asset::Cw20Token(addr) = msg.asset_info.clone() {
```

Comment:
Not sure this is necessary as the contract does not implement `Receive` for CW20 tokens

Reply:
For validating denom, `get_verified_asset` function is called which is checking if `SEND_CW20_ACTION` permission is set for the cw20 token. To avoid unnecessary permission, might need to add another trait which is exactly the same as `get_verified_asset` except the permission check.